### PR TITLE
Patersonc/add lab cip

### PR DESCRIPTION
**Pull request by patersonc**

### CODE REVIEW

**Comment by patersonc** [comment]
I'm done with all my faffing with this PR now... Sorry for the noise.

Hopefully I haven't missed anything.

File: test-configs.yaml [hunk 1]
```diff
@@ -1027,11 +1020,76 @@ device_types:
...
+      - whitelist: {defconfig: ['multi_v7_defconfig', 'shmobile_defconfig']}
+
+  r8a774a1-hihope-rzg2m-ex:
+    mach: renesas
+    class: arm64-dtb
+    boot_method: uboot
+    filters:
+      - whitelist: {defconfig: ['defconfig']}
+
+  r8a774a1-hihope-rzg2n-ex:
```
Comment (gctucker) [inline test-configs.yaml:1053]: Typo: (b1, not a1): `r8a774b1-hihope-rzg2n-ex`

**Comment by gctucker** [comment]
Skipping until the typo is fixed as it causes the YAML configuration to fail to load.

File: test-configs.yaml (hunk 1 above)
Comment (patersonc) [reply test-configs.yaml:1053]: Sigh. Thanks!

Comment (patersonc) [reply test-configs.yaml]: And fixed.

**Comment by gctucker** [comment]
Tested OK on staging:
https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200616.1/plan/baseline/

@patersonc Now, before the device types are merged and enabled in production, they should have upstream support in LAVA.  Are you OK with doing that, if they don't already?

**Comment by patersonc** [comment]
> Tested OK on staging:
> https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200616.1/plan/baseline/
> 
> @patersonc Now, before the device types are merged and enabled in production, they should have upstream support in LAVA. Are you OK with doing that, if they don't already?

It's on my todo list. Currently the device-templates are on CIP's [lava-docker](https://gitlab.com/cip-project/cip-testing/lava-docker/-/tree/cip-lava-docker/lava-master/device-types).

**Comment by gctucker** [comment]
Thanks, please put a comment with links to merge requests in LAVA's Gitlab if you make any so we can track progress from here.

**Comment by patersonc** [comment]
https://git.lavasoftware.org/lava/lava/-/merge_requests/1235

**Comment by gctucker** [comment]
Thanks, I've approved it.  I'll leave it until Monday for others (Rémi) to get a chance to look at it just in case.

Sounds like the CPI lab should get enabled in kernelci.org next week in any case :)

**Comment by patersonc** [comment]
Thanks

**Comment by gctucker** [comment]
Merged device types in upstream LAVA, this should be ready to be merged too.

File: test-configs.yaml [hunk 2]
```diff
@@ -1030,11 +1020,76 @@ device_types:
...
+
+  r8a7745-iwg22d-sodimm:
+    mach: renesas
+    class: arm-dtb
+    boot_method: uboot
+    dtb: 'r8a7745-iwg22d-sodimm-dbhd-ca.dtb'
+    filters:
+      - whitelist: {defconfig: ['multi_v7_defconfig', 'shmobile_defconfig']}
+
+  r8a77470-iwg23s-sbc:
```
Comment (gctucker) [inline test-configs.yaml:1039]: There's a problem with this device type, it's failing to boot with a ramdisk, both in health checks and in kernelci jobs:
https://lava.ciplatform.org/scheduler/job/19319
https://lava.ciplatform.org/scheduler/device_type/r8a77470-iwg23s-sbc?dt_length=25&dt_search=kernelci#dt_

So it's probably best to drop it from this PR initially, to only add devices that are known to be working well, and add it when the issue has been fixed (upstream kernel? bootloader?).

Comment (patersonc) [reply test-configs.yaml:1039]: I'm happy to drop this device for now. Please double check that the other devices are working okay for you though.

File: test-configs.yaml [hunk 3]
```diff
@@ -1030,11 +1020,76 @@ device_types:
     filters:
       - whitelist: {defconfig: ['defconfig']}
 
+  r8a7743-iwg20d-q7:
```
Comment (gctucker) [inline test-configs.yaml:1023]: This one doesn't seem happy either:
https://lava.ciplatform.org/scheduler/device_type/r8a7743-iwg20d-q7?dt_length=25&dt_search=kernelci#dt_

File: test-configs.yaml [hunk 4]
```diff
@@ -1030,11 +1020,76 @@ device_types:
...
+    mach: renesas
+    class: arm-dtb
+    boot_method: uboot
+    dtb: 'r8a7743-iwg20d-q7-dbcm-ca.dtb'
+    filters:
+      - whitelist: {defconfig: ['multi_v7_defconfig', 'shmobile_defconfig']}
+
+  r8a7745-iwg22d-sodimm:
```
Comment (gctucker) [inline test-configs.yaml:1031]: Ditto: https://lava.ciplatform.org/scheduler/device_type/r8a7745-iwg22d-sodimm?dt_length=25&dt_search=kernelci#dt_

**Comment by patersonc** [comment]
Hi @gctucker ,

I've submitted a PR to LAVA that should fix the issue where the Kernel never loads properly on the RZ/G1 platforms:
https://git.lavasoftware.org/lava/lava/-/merge_requests/1245

Sorry for the noise, I usually only use uImage, not zImage.

I think the ramfs issue is still there though so I'll just drop those boards for now.

Regards, Chris

**Comment by gctucker** [comment]
@patersonc Please drop from this PR the devices that aren't passing baseline tests so the lab can be enabled with the working ones.  The other devices can be added later when they're working as expected.

Looking at the results on the frontend, these are good:
* beaglebone-black https://staging.kernelci.org/test/plan/id/5ef46ab0be6fcb8cedf56805/
* qemu https://staging.kernelci.org/test/plan/id/5ef469ff80cd488e99f56834/
* r8a774a1-hihope-rzg2m-ex https://staging.kernelci.org/test/plan/id/5ef4743fe0a119f17af567f8/
* r8a774b1-hihope-rzg2n-ex https://staging.kernelci.org/test/plan/id/5ef4743b9de81d6694f567f4/
* r8a774c0-ek874 https://staging.kernelci.org/test/plan/id/5ef47448e0a119f17af5680b/

and these are bad:
* r8a7745-iwg22d-sodimm https://staging.kernelci.org/test/plan/id/5ef46b1b2c910be58bf5681d/
* r8a77470-iwg23s-sbc https://staging.kernelci.org/test/plan/id/5ef46b15a7d25c9508f567fe/

**Comment by patersonc** [comment]
Hi @gctucker 

I've rebased on the latest master and removed the RZ/G1 devices.
Hopefully all will be well now.

Chris

**Comment by gctucker** [comment]
Tested OK on staging:
https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200626.0/plan/baseline/

All the devices in this PR are passing all the baseline tests.

**Pull request merged**